packages/components/src/templates/next/components/complex/DGSSearchableTable/DGSSearchableTable.tsx

```diff
@@ -0,0 +1,88 @@
+"use client"
+
+import { useEffect, useMemo, useState } from "react"
+
+import type { DGSResponse, DGSSearchableTableProps } from "~/interfaces"
+import type { SearchableTableClientProps } from "~/templates/next/components/internal/SearchableTable/SearchableTableClient"
+import { SearchableTableClient } from "~/templates/next/components/internal/SearchableTable/SearchableTableClient"
+import { fetchDataFromDGS } from "./fetchDataFromDGS"
+
+export const DGSSearchableTable = ({
+  title,
+  dgsResourceId,
+  headers,
+  site,
+  LinkComponent,
+}: DGSSearchableTableProps) => {
+  const [isLoading, setIsLoading] = useState(true)
+  const [isError, setIsError] = useState(false)
+  const [data, setData] = useState<DGSResponse | null>(null)
+
+  const labels = useMemo(() => headers.map((header) => header.label), [headers])
+
+  const fieldKeys = useMemo(
+    () => headers.map((header) => header.key),
+    [headers],
+  )
+
+  useEffect(() => {
+    async function fetchAllData() {
+      try {
+        // First fetch (default limit from DGS API)
+        const initialData = await fetchDataFromDGS({
+          resourceId: dgsResourceId,
+          fields: fieldKeys,
+        })
+
+        const numberOfRecords = initialData.result.total
+
+        // If there are no records, set empty data and return early
+        if (!numberOfRecords || numberOfRecords === 0) {
+          throw new Error("No records found")
+        }
+
+        const response = await fetchDataFromDGS({
+          resourceId: dgsResourceId,
+          fields: fieldKeys,
+          limit: numberOfRecords,
+        })
+
+        setData(response)
+      } catch {
+        setIsError(true)
+      } finally {
+        setIsLoading(false)
+      }
+    }
+
+    void fetchAllData()
+  }, [])
```

Comment:
- refer to PR description on why we call the API twice
- putting in a useEffect this way also because otherwise it will fail during SSG buildtime